rebasehelper/helpers/koji_helper.py

```diff
@@ -307,36 +308,80 @@ def download_build(cls, session, build_id, destination):
+                if local_path not in rpms:
                     url = '/'.join([
                         session.opts['topurl'],
                         'packages',
                         build['package_name'],
                         build['version'],
                         build['release'],
-                        'data',
-                        'logs',
                         pkg['arch'],
-                        logname])
+                        filename])
                     DownloadHelper.download_file(url, local_path)
-                    logs.append(local_path)
-            filename = '.'.join([pkg['nvr'], pkg['arch'], 'rpm'])
+                    rpms.append(local_path)
+        else:
+            pkg = [pkg for pkg in packages if pkg['arch'] == 'src'][0]
+            filename = '.'.join([pkg['nvr'], 'src', 'rpm'])
             local_path = os.path.join(destination, filename)
-            if local_path not in rpms:
-                url = '/'.join([
-                    session.opts['topurl'],
-                    'packages',
-                    build['package_name'],
-                    build['version'],
-                    build['release'],
-                    pkg['arch'],
-                    filename])
-                DownloadHelper.download_file(url, local_path)
-                rpms.append(local_path)
+            url = '/'.join([
+                session.opts['topurl'],
+                'packages',
+                build['package_name'],
+                build['version'],
+                build['release'],
+                pkg['arch'],
+                filename])
+            DownloadHelper.download_file(url, local_path)
+            rpms.append(local_path)
         return rpms, logs
+
+    @classmethod
+    def get_old_build_info(cls, package_name, package_version):
+        """Gets latest build info from koji.
+
+        Args:
+            package_name (str): Package name from specfile.
+            package_version (str): Package version from specfile.
+
+        Returns:
+            tuple: Koji version, koji build id, package version, package full version.
+
+        """
+        if cls.functional:
+            session = KojiHelper.create_session()
+            koji_version, koji_build_id = KojiHelper.get_latest_build(session, package_name)
+            if koji_version:
+                if koji_version != package_version:
+                    logger.warning('Version of the latest Koji build (%s) with id (%s) '
+                                   'differs from version in SPEC file (%s)!',
+                                   koji_version, koji_build_id, package_version)
+                package_version = package_full_version = koji_version
+                return koji_build_id, package_version, package_full_version
+            else:
+                logger.warning('Unable to find the latest Koji build!')
+        else:
+            logger.warning('Unable to get the latest Koji build!')
```

Comment:
What happens if any of these two error cases occurs? The method is expected to return a 3-tuple.

Reply:
I guess some kind of new `KojiError` exception should be raised, as no info is provided at all.

Reply:
You'll get `ValueError` when trying to unpack `None` (the implicit return value) into `koji_build_id, package_version, package_full_version`.
The old code expects `koji_build_id` to be either build id or `None`, so you could just return `None, None, None` and it should continue working, unless I'm missing something.